demos/racetrack/Driver.cpp

```diff
@@ -123,27 +129,31 @@ void MyDisplayHandler(unsigned long windowID, tKeyboardModifier mod, char key) /
 	{
 		case 'r':
 			// TODO: Reset to start state
+			r->Reset(s);
 			break;
 			// TODO: Make appropriate movements
 			// TODO: Add support for WASD here
 		case kUpArrow: // y velocity goes up
 			std::cout << "Up arrow!" << std::endl;
-			s.yVelocity = s.yVelocity + 1;
-			if (s.yVelocity > 5){
-				s.yVelocity = 5;
-			}
-			std::cout << s.yVelocity << std::endl;
+			v.xDelta = 0;
+			v.yDelta = -1;
+			r->ApplyAction(s, v);
 			break;
 		case kDownArrow:
 			std::cout << "Down arrow!" << std::endl;
-			s.yVelocity = s.yVelocity - 1;
-			if (s.yVelocity < 0){
-				s.yVelocity = 0;
-			}
+			v.xDelta = 0;
+			v.yDelta = 1;
+			r->ApplyAction(s, v);
 			break;
 		case kLeftArrow:
+			v.xDelta = -1;
+			v.yDelta = 0;
+			r->ApplyAction(s, v);
 			break;
 		case kRightArrow:
```

Comment:
You can put two case statements together here -- for kRightArrow and 'w'. They will then run the same code. eg:
case kRightArrow:
case 'w':
<....>
break;